cct/module.py

```diff
@@ -273,19 +275,18 @@ def __init__(self, name, chksum, url):
         self.name = name
         self.chksum = chksum
         self.url = self.replace_variables(url) if '$' in url else url
-        self.filename = os.path.basename(url)
+        self.filename = name
         self.path = None
 
     def fetch(self, directory):
-        logger.debug("fetch to dir %s" % inspect.getmodule(self.__class__).__name__)
-        logger.debug("Fetching %s as a resource for module %s" % (self.url, self.name))
-
         self.path = os.path.join(directory, self.filename)
 
         if self.check_sum():
-            logger.debug("Using cached artifact for %s" % self.name)
+            logger.info("Using cached artifact for %s" % self.name)
             return
 
+        logger.info("Fetching %s as a resource for module %s" % (self.url, self.name))
```

Comment:
Sometimes we talk about artifacts, sometimes resources. We need to use one name.